Articles/Blog/2023-10-fear-of-large-codes.md

```diff
@@ -45,35 +44,48 @@ It is natural to ask at this point: why is this fear of large codes so common?
 I would argue that this approach must change for software to be sustainable, accessible, and maintainable.
 Software development should be a first-class entity in a research effort focusing on computational sciences.
 Obviously, reconciling this with the actual research questions to address and balancing it all out, budget-wise, is hard.
 But, for the sake of the argument, what could happen if this were the case?
 In other words, what could happen if a non-trivial part of a research budget was explicitly dedicated to the software development required?
-For graduate students, this would have a key benefit: it would allow students to learn many skills along the way, e.g. team development, testing, and software quality, that would later prove critical for the job market.
-After all, many graduate students and junior researchers are the computational scientists of the future.
-More broadly, for a research idea or a proposal, I see two critical benefits: (a) solid, robust software would decrease the chances of doing something incorrectly and, therefore, increase the trustworthiness of the results and benefit scientific reproducibility; (b) any research code developed from the ground up with good practices would less likely be forgotten and more likely to become the starting point for a production code.
+Broadly speaking, I would argue that it could benefit the scientific community as a whole.
+First, it would allow graduate students and junior researchers to learn many skills along the way, e.g. team development, testing, and software quality, that would later prove critical for the job market---after all, many of them are the computational scientists of the future.
+Second, developing solid and robust software would decrease the chances of doing something incorrectly and, therefore, increase the trustworthiness of the results, which is critical for scientific reproducibility.
+Finally, any research code developed from the ground up with good practices would less likely be forgotten and more likely become the starting point for a production code (or a stepping stone for other work).
+# For graduate students, this would have a key benefit: it would allow students to learn many skills along the way, e.g. team development, testing, and software quality, that would later prove critical for the job market.
+# After all, many graduate students and junior researchers are the computational scientists of the future.
+# More broadly, for a research idea or a proposal, I see two critical benefits: (a) solid, robust software would decrease the chances of doing something incorrectly and, therefore, increase the trustworthiness of the results and benefit scientific reproducibility; (b) any research code developed from the ground up with good practices would less likely be forgotten and more likely to become the starting point for a production code.
```

Comment:
Is this level1 heading formatting desired? Is this supposed to be commented out?

Reply:
commented out ! just fixed